### PR TITLE
Support for Generating to Composer Packages

### PR DESCRIPTION
**Pull request by astorm**
Also include 

- Updating to a semi-modern version of PHPUnit

### CODE REVIEW

**Comment by enderandpeter** [comment]
Heck yeah! This tool is already wonderful, but it would be spectacular if it would also help you make modules as Composer packages.

File: build.xml [hunk 1]
```diff
@@ -28,29 +28,34 @@
                 <exclude name="**/phpspec/**" />
                 <exclude name="**/phpunit/**" />
                 <exclude name="**/sebastian/**" />
-                <exclude name="**/symfony/**" />
-            </fileset>					
+                <exclude name="**/symfony/yaml/**" />
```
Comment (astorm) [inline build.xml:31]: Seems like vendor/symfony/polyfill-ctype/bootstrap.php gets added to the composer autoloader, which is used by the phar, so this needs to be here.

File: composer.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     },
 
     "require-dev": {
-        "phpunit/phpunit": "4.8.*",
+        "phpunit/phpunit": "^6.5",
         "phing/phing": "2.*",
```
Comment (astorm) [inline composer.json:15]: Doing the smallest possible PHPUnit upgrade that will allow us to solve our warnings and still have PHP 7.{0,1,2,3,4} support.

File: modules/pulsestorm/magento2/cli/fix_direct_om/module.php [hunk 3]
```diff
@@ -323,22 +323,24 @@ function prepareResultsIfCreateFactoryIsNeeded($array)
...
-        replaceObjectManager($file, $array, $tokens_all);                             
-        performInjection($file, $array);        
+        replaceObjectManager($file, $array, $tokens_all);
+        performInjection($file, $array);
     }
 }
 
-function getBaseMagentoDirFromFile($dir)
+function getBaseMagentoDirFromFile($dir,$skipRealPath=false)
 {
```
Comment (astorm) [inline modules/pulsestorm/magento2/cli/fix_direct_om/module.php:334]: Added this for testing so we can use a fake path.

File: modules/pulsestorm/magento2/cli/generate/module/module.php [hunk 4]
```diff
@@ -9,6 +9,18 @@
 pestle_import('Pulsestorm\Xml_Library\addSchemaToXmlString');
 pestle_import('Pulsestorm\Cli\Code_Generation\templateRegistrationPhp');
 pestle_import('Pulsestorm\Magento2\Cli\Xml_Template\getBlankXml');
+pestle_import('Pulsestorm\Magento2\Cli\Library\getModuleInformation');
+
```
Comment (astorm) [inline modules/pulsestorm/magento2/cli/generate/module/module.php:13]: Composer packages usually have their code in a folder that's different from the package base, so we now have the concepts of a module directory (where's the code!) and a package directory (where's composer.json and registration.php)

File: modules/pulsestorm/magento2/cli/library/module.php [hunk 5]
```diff
@@ -17,24 +17,103 @@
...
+    list($vendor, $name) = explode('_', $module_name);
+    $information = [
         'vendor'        => $vendor,
         'short_name'    => $name,
         'name'          => $module_name,
-        'folder'        => getBaseMagentoDir() . "/app/code/$vendor/$name",
     ];
+
+
+    // we need to check the configuration for a path.  If it exists, then
```
Comment (astorm) [inline modules/pulsestorm/magento2/cli/library/module.php:86]: Lots of shenanagins to make sure we're loading the right path depending on whether the user's working out of app/code or a vendor folder.

File: modules/pulsestorm/pestle/runner/module.php [hunk 6]
```diff
@@ -145,7 +145,8 @@ function getListOfDefinedCliFunctions()
     foreach($namespaces as $namespace)
     {
         if(in_array($namespace, ['',$current_namespace,
-            'composer\autoload','guzzlehttp\promise','guzzlehttp\psr7','guzzlehttp']))
+            'composer\autoload','guzzlehttp\promise','guzzlehttp\psr7',
+            'guzzlehttp','deepcopy']))
         {
```
Comment (astorm) [inline modules/pulsestorm/pestle/runner/module.php:150]: Another namespace to ignore.  Probably should provide an API for this.

File: tests-integration/PestleTestIntegration.php [hunk 7]
```diff
@@ -3,9 +3,9 @@
 namespace Pulsestorm\Pestle\TestsIntegration;
 
 use Exception;
-use PHPUnit_Framework_TestCase;
-
-class PestleTestIntegration extends PHPUnit_Framework_TestCase
+//use PHPUnit_Framework_TestCase;
+use PHPUnit\Framework\TestCase;
+class PestleTestIntegration extends TestCase
 {
```
Comment (astorm) [inline tests-integration/PestleTestIntegration.php:9]: Huzzah! All hail PHP 5.3

**Pull request merged**